CanvasColorSpaceProposal.md

```diff
@@ -45,178 +45,172 @@ You can see this by making stripes of 1.0 and 0.0, and matching that pattern's b
+dictionary ImageDataAttributes {
+  CanvasColorSpace colorSpace = "srgb";
+  ImageDataStorageFormat storageFormat = "uint8";
+};
 
-  ImageDataColorSettings getColorSettings();
+partial interface ImageData {
+  ImageDataAttributes getAttributes();
+  ImageDataArray data;
 };
 </pre>
 
-* When using the constructor that takes an ImageDataArray parameter, the "storageType" setting is ignored.
-* createImageData() and getImageData() produce an ImageData object with the same color space as the source canvas, using an ImageDataArray of a type that is appropriate for the pixelFormat of the source canvas (smallest possible numeric size that guarantees no loss of precision).
-* putImageData() performs a color space conversion to the color space of the destination canvas.
-* Data returned by getImageData() or passed to putImageData() are assumed to be in "srgb" color space, with "unorm8" encoding.
+An ImageData's data attribute will be of the type indicated by the its storageFormat.
 
-### Limitations
-* toDataURL and toBlob may be lossy, depending on the file format, when used on a canvas that has an encoding other than "unorm8". Possible future improvements could solve or mitigate this issue by adding more file formats or adding options to specify the resource color space.
+Note that the storage formats are different from the pixel formats for 2D Canvas and WebGL pixel formats. This reflects the fact that there does not exist a Float16Array, although Float16 is a useful format for graphics buffers.
+
+### ImageBitmap
+
+ImageBitmaps are opaque bitmap objects that can be used by several consumers. ImageBitmaps should internally store their contents in the highest fidelity format possible.
+
+For example, if an ImageBitmap is created from a wide-color-gamut high-bit-depth image and is then drawn to a wide-color-gamut 2D canvas, the result should reflect the original image. In particular, the image should not have been clamped to sRGB or quantized to 8 bits anywhere along the way.
+
+TODO: createImageBitmap has a colorSpaceConversion flag that can be "none" or "default". The most useful behavior is "respect the color space and convert appropriately when drawing to a canvas", which is the behavior of "default" on some browsers. A less useful but sometimes desirable behavior is "ignore the color space, and reinterpret the pixels as though they were in sRGB", which is the behavior of "none". This behavior should be more clearly specified.
 
 ### Adoption
+
 Lack of color management and color interoperability is a longstanding complaint about the canvas API.
 Authors of games and imaging apps are expected to be enthusiastic adopters.
 
-## Unresolved Issues
+## Issues
 
-* Should we support custom color spaces based on ICC profiles? Would offer ultimate flexibility. Would be hard to make implementations as efficient as built-in color spaces, in particular for implement linearPixelMath for profiles that have arbitrary transfer curves.
+Issues:
```

Comment:
How will a web developer determine whether advanced color management is necessary?  

On an SDR monitor, using the proposed options incurs unnecessary texture space (for float16 textures) and processing time to tonemap from the source to destination color gamuts/spaces.

Reply:
The color gamut media queries can be used to determine if there is a benefit to using p3 or rec2020.
https://developer.mozilla.org/en-US/docs/Web/CSS/@media/color-gamut

(A similar media query should be added for HDR, but hasn't, to my knowledge).